### PR TITLE
Create simple wrapper that starts both http and https server

### PR DESCRIPTION
**Pull request by tlbdk**

### CODE REVIEW

File: src/http/web-server.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return [...this.httpServer.getJsonRequests(), ...this.httpsServer.getJsonRequests()]
+  }
+
+  public getTextRequests(): HttpTextRequest[] {
+    return [...this.httpServer.getTextRequests(), ...this.httpsServer.getTextRequests()]
+  }
+
+  public getRequests(): HttpRequest[] {
+    return [...this.httpServer.getRequests(), ...this.httpsServer.getRequests()]
+  }
```
Comment (mex) [inline src/http/web-server.ts]: Shouldn't these be ordered chronologically and not just concatenated?

File: src/http/http-server-base.ts [hunk 2]
```diff
@@ -94,7 +94,8 @@ export class HttpServerBase<T extends http.Server | https.Server> {
       method: req.method,
       url: req.url,
       headers: headers,
-      body: await readBody(req)
+      body: await readBody(req),
+      receivedAt: new Date()
```
Comment (mex) [inline src/http/http-server-base.ts]: We should probably strip this value in the `get[...]Requests` as it will otherwise mess with snapshots.

Comment (tlbdk) [reply src/http/http-server-base.ts]: I could also change it to a number instead?

Comment (mex) [reply src/http/http-server-base.ts]: Yeah, that should work as well.

Comment (tlbdk) [reply src/http/http-server-base.ts]: Done

File: src/http/http-server-base.ts [hunk 3]
```diff
@@ -4,33 +4,36 @@ import net from 'net'
...
   readBody
 } from './http-common'
 
 export class HttpServerBase<T extends http.Server | https.Server> {
-  public listenPort = 0
+  public listenPort: number
   public listenUrl = ''
   private baseUrl: string
   protected httpServer: T
   protected requests: HttpRequest[] = []
```
Comment (mex) [inline src/http/http-server-base.ts]: We can add the `requests` array as an option so you can reference it back to a request array in the WebServer class.

**Pull request merged**